### PR TITLE
UI and logic fixes for bend canvas

### PR DESCRIPTION
**Pull request by Marr11317**
Resolves: no issue in issue tracker

Ui fixes for the bendcanvas.

It was like that:

image

and is now like that:
image

and working in dark mode:
image

To summarize:
 - it was not antialisased,
 - there was no dark mode
 - Due to using integers instead of floating points values, the graph was not well centered.
 - the gray line between dots was not clearly visible
 - square felt old-style in my opinion. Tell me if you think I should revert it.

on the logic side, the comparison was done weirdly. Suppose this score (imported with musicxml, because not it is not doable in MS):
```
...
            <Bend>
                <point time="0" pitch="0" vibrato="0"/>
                <point time="14" pitch="101" vibrato="0"/>
                <point time="60" pitch="100" vibrato="0"/>
                </Bend>
...
```
when showing in the bandcanvas, it would show this:
image

Which made it impossible to remove the point as points could only be placed at intersections.Clicking on it to remove it gives this:
image

or, when making the inspector bigger:
image

The same thing after this PR centers the point at an intersection (without modifying the value), and makes it deletable and modifiable.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- N/A I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by Howard-C** [comment]
I like this, but you'll have to change tremolo bar canvas too, since they're basically the same. Or further, go make bend canvas and tremolo bar canvas subclasses of a new base class, like `SquareCanvas` or whatever. In this case the code will be much simpler.

File: mscore/bendcanvas.cpp [hunk 1]
```diff
@@ -40,62 +42,75 @@ BendCanvas::BendCanvas(QWidget* parent)
...
+      // let half a column of margin around
+      const qreal leftPos = columnWidth * .5; // also left margin
+      const qreal topPos = rowHeight * .5;    // also top margin
+      const qreal rightPos = w - leftPos; // right end position of graph
+      const qreal bottomPos = h - topPos; // bottom end position of graph
+
+      QPainter painter(this);
+      painter.setRenderHint(QPainter::Antialiasing);
+      // #2c2c2c is musescore's "dark color", so a little brighter
+      painter.fillRect(rect(), preferences.isThemeDark() ? QColor("#3c3c3c") : Qt::white);
```
Comment (Howard-C) [inline mscore/bendcanvas.cpp:63]: `#2c2c2c` or `#3c3c3c`? If it's the general dark colour, shouldn't there be a variable to represent it?

Comment (Marr11317) [reply mscore/bendcanvas.cpp:63]: It is in `palette_dark_fusion.json`
I wonder if that is accessible. Let me check.

**Comment by Marr11317** [comment]
> 
> 
> I like this, but you'll have to change tremolo bar canvas too, since they're basically the same. Or further, go make bend canvas and tremolo bar canvas subclasses of a new base class, like `SquareCanvas` or whatever. In this case the code will be much simpler.

I didn't know about the tremolo bar canvas. I'll check it out.

**Comment by Marr11317** [comment]
It was weirder then expected to match the two, but it worked out:
Before:
image

After:
image

**Comment by Howard-C** [comment]
I was hoping you'd think of a better name than `SquareCanvas`, but good job!

**Pull request merged**